joomla/joomla-cms
# [4.2] NPM build tools cleanup

Pull Request for Issue # .

### Summary of Changes
- remove some I/O operations on both the JS and SCSS minification (eg use the code output directly)
- Ran `npm lint:js -- --fix` and committed the changes
  - Removed the IIFE from [build/media_source/com_content/js/articles-status.es6.js](https://github.com/joomla/joomla-cms/compare/4.2-dev...dgrammatiko:joomla-cms:4.2-dev-buildtools?expand=1#diff-4afb4f910136391957e62b8afcb5f0d6b240ee226cd69b4960777bc15ddfe668) as the file is correctly loaded either as module or when in `nomodule` it's using the defer attribute, thus adding the listener for `DOMContentLoaded` is pure overhead. At that point the DOM is already there fully parsed. This needs to be applied **ON EVERY SCRIPT**


### Testing Instructions
On a fresh installation of Joomla (ie `git clone`) do the composer install and then the npm install. Copy the media folder somewhere in your hardisk. (check the time needed)
Now clone this PR and do both the composer and npm installations. Once done do a folder compare of the current media folder against the folder that you saved in the previous step. You should have the same files apart from the `build/media_source/com_content/js/articles-status.es6.js` which will have different content (expected).

Compare the times and do a brief check that both front end and back still work as expected


### Actual result BEFORE applying this Pull Request

<img width="555" alt="Screenshot 2022-07-04 at 18 36 53" src="https://user-images.githubusercontent.com/3889375/177193375-f2bedfa5-8d44-4d77-b4e3-e74290e05afc.png">


### Expected result AFTER applying this Pull Request

<img width="478" alt="Screenshot 2022-07-04 at 18 35 50" src="https://user-images.githubusercontent.com/3889375/177193391-9c556f89-404f-4c51-a449-91e6c9b20e52.png">


### Documentation Changes Required

NO

@HLeithner have somebody merge this so I could do the changes in the v5.0. Thanks
